lib/loggerator/railtie/adapter.rb

```diff
@@ -0,0 +1,24 @@
+module Loggerator
+  module Railtie
+    class Adapter < ::Rails::Railtie
+      config.before_configuration do
+        Rails.application.middleware.insert_after ActionDispatch::RequestId, Loggerator::Middleware::RequestStore
+        Rails.application.middleware.swap         ActionDispatch::RequestId, Loggerator::Middleware::RequestID
+      end
+
+      config.before_initialize do
+        [ ActionView::Base,
+          ActiveRecord::Base,
+          ActionMailer::Base,
+          ActionController::Base ].each do |c|
+
+          c.include Loggerator
+        end
+      end
+
+      config.after_initialize do
+        Loggerator::Railtie::Helper.setup(Rails.application) if Loggerator::Railtie::Helper.subscribe?
```

Comment:
I suggest putting the `subscribe?` logic into the `setup` method.  Simplifies the code to call and keeps the `subscribe?` logic inside the helper (where it's closer to the code that uses it).